File: src/MusclePy/twin_model/twin_nodes_results.py
import numpy as np

class Twin_NodesResults:
    def __init__(self):
        """Python equivalent of C# Twin_NodesResults class"""
        self.displacements = np.array([], dtype=float)  # [m] - shape (NodesCount,3)
        self.residual = np.array([], dtype=float)  # [N] - shape (NodesCount,3)
        self.reactions = np.array([], dtype=float)  # [N] - shape (NodesCount,3)

File: src/MuscleCS/TwinModel/TwinNodesResults.cs


namespace MuscleCore.TwinModel
{
	public class TwinNodesResults
	{
		public double[,] Displacements { get; set; }  //[m] - shape (NodesCount,3)

		public double[,] Residual { get; set; } //[N] - shape (NodesCount,3)
		public double [,] Reactions { get; set; } //[N] - shape (NodesCount,3)

		public TwinNodesResults()
		{
			Displacements = new double[,] { };
			Residual = new double[,] { };
			Reactions = new double[,] { };
		}
		public TwinNodesResults(double[,] displacements, double[,] residual, double[,] reactions)
		{
			Displacements = displacements;
			Residual = residual;
			Reactions = reactions;
		}
	}
}